src/DotNetWorker.Grpc/GrpcServiceCollectionExtensions.cs

```diff
@@ -38,42 +41,36 @@ public static IServiceCollection AddGrpc(this IServiceCollection services)
             services.RegisterOutputChannel();
 
             // Internal logging
-            services.AddSingleton<GrpcFunctionsHostLogWriter>();
-            services.AddSingleton<IUserLogWriter>(p => p.GetRequiredService<GrpcFunctionsHostLogWriter>());
-            services.AddSingleton<ISystemLogWriter>(p => p.GetRequiredService<GrpcFunctionsHostLogWriter>());
-            services.AddSingleton<IUserMetricWriter>(p => p.GetRequiredService<GrpcFunctionsHostLogWriter>());
-            services.AddSingleton<IWorkerDiagnostics, GrpcWorkerDiagnostics>();
+            services.TryAddSingleton<GrpcFunctionsHostLogWriter>();
+            services.TryAddEnumerable(ServiceDescriptor.Singleton<IUserLogWriter, GrpcFunctionsHostLogWriter>(p => p.GetRequiredService<GrpcFunctionsHostLogWriter>()));
```

Comment:
Different than what is done in the core worker registration, we use `TryAddEnumerable` here.